### ISSUE TITLE
Issue about reshaping a sparse matrix to a flattened vector

### ISSUE THREAD

**Issue by j20232** [scipy/scipy#14131]
I cannot reshape a sparse matrix to a flattened vector directly under Python `3.8.3` + Scipy `1.6.3`.
If I use `np.copy` before apply the `reshape` function, it's possible to convert to the completly flattened vector format.
However, it's not possible when I use `copy.copy` or `copy.deepcopy` function.
I think the output formats should be consistent with/without the copy methods. 
You can check this issue with the following snippets.

#### Reproducing code example:

```py
import numpy as np
import scipy
import copy

t0 = scipy.sparse.csr_matrix(([0], ([1], [1])), shape=(100, 100))  # shape: (100, 100)
t1 = t0.todense()  # shape: (100, 100)
t2 = t1.reshape(-1)  # shape: (1, 10000), not flattened!
t3 = np.copy(t2).reshape(-1) # shape: (10000, ), flattened!
t4 = copy.copy(t2).reshape(-1) # shape: (1, 10000), NOT flattened!
t5 = copy.deepcopy(t2).reshape(-1) # shape: (1, 10000), NOT flattened!
```

#### Scipy/Numpy/Python version information:

- Python `3.8.3`
- Numpy: `1.18.5`
- Scipy `1.6.3`.

#### Similar issues
- https://github.com/scipy/scipy/pull/3957
- https://github.com/scipy/scipy/issues/3967

Could you check this issue?
Best regards,

**Comment by ogauthe** [comment]
Hello,

When using `todense`, you obtain a `np.matrix` object, which is necessarily 2-dimensional. When calling `.reshape(-1)`, you conserve matrix type, thus 2D. When you use `np.copy`, you actually cast your `np.matrix` to a `np.ndarray`, which _can_ be 1D. Using `copy`, you conserve `np.matrix` type and cannot flatten.

To cast directly to a `np.ndarray`, you should consider `toarray` instead of `todense`:
```
t1 = t0.toarray()  # ndarray shape: (100, 100)
t2 = t1.reshape(-1)  # shape: (10000), flattened!
```

 Note that while not deprecated (yet), the official numpy documentation discourages the use of `np.matrix`. I suppose we have to weight for official depreciation before deprecating `todense`.

To summarize: what you observe is not related to `scipy.sparse` but to the different ways numpy and copy handle `np.matrix`. You may consider submitting to issue to numpy, but I suspect they just want to get rid of `np.matrix`. I advise you to avoid this type and directly use `ndarray`.

**Comment by j20232** [comment]
Thank you for your prompt response. 
OK, I'll directly use `ndarray`.
